#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Grass.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/rocks/protos/Rock.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      material1 "passive_wheel"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.12921823038870547 -0.9197937508926013 0.3705159979732733 5.563890953582692
  position -33.9589774476859 6.920607546979085 26.661659816941157
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF GROUND Solid {
  children [
    DEF GROUND_SHAPE Group {
      children [
        Pose {
          translation -8 -8 0
          children [
            Shape {
              appearance Grass {
                type "mossy"
                textureTransform TextureTransform {
                  scale 8.5 8.5
                }
              }
              geometry ElevationGrid {
                height [
                  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.0734148, 0.277421, 0.254337, 0.270417, 0.174982, 0.171189, 0.2473, 0.240992, 0.152361, 0.110575, 0.0978332, 0.126226, 0.106899, 0.0670659, 0.00953075, 0, 0, 0.499106, 0.798685, 0.613379, 0.322475, 0.287311, 0.266366, 0.418124, 0.525612, 0.451978, 0.30729, 0.0753625, 0.0480358, 0.0532171, 0.0870809, 0.0265759, 0, 0, 0.862448, 1.07618, 0.821001, 0.384595, 0.210389, 0.0983306, 0.331182, 0.347343, 0.263181, 0.13356, 0.0145599, 0.0762221, 0.100532, 0.243507, 0.159553, 0, 0, 0.75534, 0.984221, 0.864925, 0.447235, 0.0751634, 0.0222157, 0.078995, 0.102868, 0.0300013, 0.0114454, 0.00744961, 0.00984787, 0.204083, 0.520919, 0.461261, 0, 0, 0.362822, 0.603904, 0.566029, 0.340368, 0.102151, 0.0404528, 0.0668038, 0.00744951, 0.00933689, 0.0504858, 0.00744961, -0.000948593, 0.113054, 0.320988, 0.289012, 0, 0, 0.24704, 0.327251, 0.277966, 0.170757, 0.0771545, 0.0929641, 0.00744947, 0.00744951, 0.00744954, 0.00744958, 0.00744961, 0.0223858, 0.0165338, 0.0943175, 0.0460902, 0, 0, 0.205381, 0.266736, 0.119044, 0.124862, 0.0668763, 0.106241, 0.00744947, 0.00744951, 0.00744954, 0.00744958, 0.00744961, 0.00747269, 0.0113116, 0.0207513, 0.0670393, 0, 0, 0.131618, 0.109085, 0.0721561, 0.169558, 0.124192, 0.00986459, 0.00744947, 0.00744951, 0.00744954, 0.031045, 0.0443939, 0.00800915, 0.0184385, 0.0427789, 0.204962, 0, 0, 0.283209, 0.135281, 0.068453, 0.212608, 0.159615, 0.0361041, 0.00744947, 0.00744951, 0.0121282, 0.145762, 0.2493, 0.0458433, 0.0131252, 0.131277, 0.19891, 0, 0, 0.660028, 0.539195, 0.111397, 0.132591, 0.105425, 0.0289037, 0.00744947, 0.00744951, 0.00930376, 0.162809, 0.219176, 0.0290093, 0.0131459, 0.0436956, 0.0856175, 0, 0, 0.577927, 0.513154, 0.0907302, 0.0479889, 0.0704255, 0.019278, 0.00747491, 0.00932149, 0.0307641, 0.178685, 0.171126, 0.00346421, 0.0308989, 0.0948634, 0.262075, 0, 0, 0.315194, 0.205858, 0.0314991, 0.0365704, 0.127151, 0.142281, 0.00631048, 0.0559231, 0.196945, 0.333009, 0.17502, 0.00950381, 0.0639322, 0.243806, 0.282763, 0, 0, 0.233297, 0.177544, 0.0450179, 0.0682162, 0.129357, 0.0562887, 0.0190858, 0.120147, 0.389857, 0.381918, 0.0861915, -0.000387524, 0.178523, 0.482041, 0.357671, 0, 0, 0.00946765, 0.0925988, 0.208856, 0.242312, 0.359189, 0.320955, 0.189391, 0.194184, 0.352838, 0.288278, 0.142815, 0.143518, 0.27195, 0.517513, 0.504371, 0, 0, 0.037447, 0.126832, 0.314132, 0.26641, 0.227191, 0.16272, 0.107686, 0.12869, 0.0911368, 0.130659, 0.0368771, 0.108887, 0.0914181, 0.249301, 0.291638, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
                ]
                xDimension 17
                yDimension 17
              }
            }
          ]
        }
        Shape {
          appearance Grass {
            type "mossy"
            textureTransform TextureTransform {
              scale 250 250
            }
          }
          geometry Plane {
            size 500 500
          }
        }
      ]
    }
  ]
  boundingObject USE GROUND_SHAPE
}
Rock {
  translation -0.866326 -1.02326 0
  rotation 0 0 1 1.5708
}
Rock {
  translation -0.212539 1.7806 -0.037219
  rotation 0 1 0 0.0980463
  name "rock (1)"
  type "flat"
}
Rock {
  translation 0.814429 0.889895 0
  rotation 0 0 1 1.309
  name "rock (2)"
  type "flat"
}
Rock {
  translation -2.2847 2.36097 0
  rotation -0.1325999849101226 0.1325999849101226 0.9822598882188313 -1.5886953071795862
  name "rock (3)"
  type "flat"
}
Rock {
  translation -2.19116 0.215003 0
  rotation 0 0 1 1.5708
  name "rock (4)"
  type "flat"
}
DEF AGENT Robot {
  translation 4.18002 2.82006 0.07
  rotation 0 0 -1 1.5708
  children [
    TouchSensor {
      translation 0.01 0 0.75
      children [
        DEF TOUCHSENSOR_BODY Shape {
          geometry Box {
            size 0.005 0.02 1.5
          }
        }
      ]
      name "touchsensor_front"
      boundingObject USE TOUCHSENSOR_BODY
    }
    TouchSensor {
      translation -0.01 0 0.75
      children [
        DEF TOUCHSENSOR_BODY Shape {
          geometry Box {
            size 0.005 0.02 1.5
          }
        }
      ]
      name "touchsensor_back"
      boundingObject USE TOUCHSENSOR_BODY
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.1 -0.11 -0.02
      }
      device [
        RotationalMotor {
          name "wheel4"
          maxVelocity 15
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.09999999999999962 -0.10999999999999943 -0.019999999999999206
        rotation 1 8.886706994834418e-14 -6.58260054343521e-14 1.5707999999959477
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid4(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.1 0.11 -0.02
      }
      device [
        RotationalMotor {
          name "wheel3"
          maxVelocity 15
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation -0.10000000000000014 0.11000000000000032 -0.019999999999999536
        rotation 1 3.680289963937376e-13 -3.5091502002559885e-14 1.5707999999965547
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid3(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.1 -0.11 -0.02
      }
      device [
        RotationalMotor {
          name "wheel2"
          maxVelocity 15
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation 0.10000000000000007 -0.10999999999999943 -0.02000000000000003
        rotation 1 2.4532649610291924e-14 -7.361757495056401e-14 1.5707999999964595
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid2(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.1 0.11 -0.02
      }
      device [
        RotationalMotor {
          name "wheel1"
          maxVelocity 15
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.09999999999999959 0.11000000000000032 -0.019999999999999803
        rotation 1 4.346007943763741e-13 -5.024286640189296e-15 1.570799999995858
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    Solid {
      translation 5.15213e-15 1.74204e-21 1.55
      rotation 0 0 1 1.01503e-06
      children [
        Display {
          width 10
          height 10
        }
        Camera {
          fieldOfView 2.356
          width 640
          height 240
          recognition Recognition {
            segmentation TRUE
          }
        }
        DEF CAMERA_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "camera_box"
      boundingObject USE CAMERA_SHAPE
    }
    DEF ROBOT Group {
      children [
        Pose {
          translation 0 0 0.75
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.02 0.02 1.5
              }
            }
          ]
        }
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.3 0.2 0.05
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.145 0.05 0.03
      rotation 0 0 1 0.523599
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds1"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0.145 0 0.03
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds2"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0.145 -0.05 0.03
      rotation 0 0 1 -0.523595307179586
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds3"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0.07 -0.09 0.03
      rotation 0 0 1 -0.785395307179586
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds4"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0 -0.09 0.03
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds5"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation -0.07 -0.09 0.03
      rotation 0 0 1 -2.094395307179586
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds6"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation -0.14 -0.05 0.03
      rotation 0 0 -1 2.618
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds7"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation -0.14 0 0.03
      rotation 0 0 1 3.14159
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds8"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation -0.14 0.05 0.03
      rotation 0 0 1 2.61799
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds9"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation -0.07 0.09 0.03
      rotation 0 0 1 2.0944
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds10"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0 0.09 0.03
      rotation 0 0 1 1.5708
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds11"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
    DistanceSensor {
      translation 0.07 0.09 0.03
      rotation 0 0 1 0.785398
      children [
        DEF DS_SHAPE Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds12"
      boundingObject USE DS_SHAPE
      lookupTable [
        0 0 0
        0.5 100 0
      ]
      aperture 2
    }
  ]
  boundingObject USE ROBOT
  physics Physics {
  }
  controller "<extern>"
  supervisor TRUE
}
